path/spanning_tree.go

```diff
@@ -5,101 +5,166 @@
 package path
 
 import (
+	"container/heap"
+	"math"
 	"sort"
 
 	"github.com/gonum/graph"
 	"github.com/gonum/graph/concrete"
-	"github.com/gonum/graph/internal"
 )
 
-// EdgeListerGraph is an undirected graph than returns its complete set of edges.
-type EdgeListerGraph interface {
+// UndirectedWeighter is an undirected graph that returns distinct edge weights.
```

Comment:
> Though I could clarify in Prim and Kruskal that they find "a minimum spanning tree, which will be the minumum spanning tree if the edge weights are distinct" and drop the distinct part from the interface doc comment.

The wording you propose is nice, not much of a hand-holding but still transmits the key message.